pkg/urbit/noun/nock.c

```diff
@@ -1838,6 +1838,23 @@ _n_kale(u3_noun a)
   return a;
 }
 
+/* _n_sane(): check for stack overflow
+*/
+static inline void
+_n_sane(c3_ys off_ys)
+{
+  if ( !off_ys ) {
+    if( !(u3R->cap_p > u3R->hat_p) ) {
+      u3m_bail(c3__meme);
+    }
+  }
+  else {
+    if( !(u3R->cap_p < u3R->hat_p) ) {
+      u3m_bail(c3__meme);
+    }
+  }
+}
```

Comment:
These changes (all of commit 4bd4e8a4a9) are irrelevant in the presence of the guard page.

I'm tempted to leave existing stack overflow checks (not here), behind a `#define` that also toggles the guard page implementation. Then anyone embedding u3 (currently no one) can choose how to handle overflow checks.

Reply:
Reverted `4bd4e8a`.

Reply:
Reverting `4bd4e8a` leads to `nock_tests` failing on my machine.